experimental/traffic-portal/src/app/api/physical-location.service.ts

```diff
@@ -23,33 +22,75 @@ import { APIService } from "./base-api.service";
  */
 @Injectable()
 export class PhysicalLocationService extends APIService {
-	public async getPhysicalLocations(idOrName: number | string): Promise<PhysicalLocation>;
-	public async getPhysicalLocations(): Promise<Array<PhysicalLocation>>;
+	public async getPhysicalLocations(): Promise<Array<ResponsePhysicalLocation>>;
+	public async getPhysicalLocations(nameOrID: string | number): Promise<ResponsePhysicalLocation>;
+
 	/**
-	 * Gets one or all PhysicalLocations from Traffic Ops
+	 * Gets an array of physicalLocations from Traffic Ops.
 	 *
-	 * @param idOrName Either the integral, unique identifier (number) or name (string) of a single PhysicalLocation to be returned.
-	 * @returns The requested PhysicalLocation(s).
+	 * @param nameOrID If given, returns only the PhysicalLocation with the given name
+	 * (string) or ID (number).
+	 * @returns An Array of PhysicalLocation objects - or a single PhysicalLocation object if 'nameOrID'
+	 * was given.
 	 */
-	public async getPhysicalLocations(idOrName?: number | string): Promise<PhysicalLocation | Array<PhysicalLocation>> {
+	public async getPhysicalLocations(nameOrID?: string | number): Promise<Array<ResponsePhysicalLocation> | ResponsePhysicalLocation> {
 		const path = "phys_locations";
-		let prom;
-		if (idOrName !== undefined) {
+		if(nameOrID) {
 			let params;
-			switch (typeof idOrName) {
+			switch (typeof nameOrID) {
 				case "string":
-					params = {name: idOrName};
+					params = {name: nameOrID};
 					break;
 				case "number":
-					params = {id: String(idOrName)};
+					params = {id: String(nameOrID)};
 			}
-			prom = this.get<[PhysicalLocation]>(path, undefined, params).toPromise().then(
-				r => r[0]
-			);
-		} else {
-			prom = this.get<Array<PhysicalLocation>>(path).toPromise();
+			const r = await this.get<[ResponsePhysicalLocation]>(path, undefined, params).toPromise();
+			return {...r[0], lastUpdated: new Date((r[0].lastUpdated as unknown as string).replace("+00", "Z"))};
+
 		}
-		return prom;
+		const physicalLocations = await this.get<Array<ResponsePhysicalLocation>>(path).toPromise();
+		return physicalLocations.map(
+			d => ({...d, lastUpdated: new Date((d.lastUpdated as unknown as string).replace("+00", "Z"))})
+		);
+	}
+
+	/**
+	 * Replaces the current definition of a physicalLocation with the one given.
+	 *
+	 * @param physicalLocation The new physicalLocation.
+	 * @returns The updated physicalLocation.
```

Comment:
The object is `[Response]PhysicalLocation` and the concept it represents is a "Physical Location," either of which is more appropriate than `physicalLocation`.